lib/livebook/runtime/erl_dist/manager.ex

```diff
@@ -152,12 +184,14 @@ defmodule Livebook.Runtime.ErlDist.Manager do
 
   @impl true
   def terminate(_reason, state) do
-    Code.compiler_options(ignore_module_conflict: state.initial_ignore_module_conflict)
+    if state.cleanup_on_termination do
```

Comment:
Apart from unloading modules, when it comes to global configuration we normally store it in the state and revert back on termination. With a single manager this works fine, but once there are multiple managers we don't want to revert the configuration, because there may be other managers depending on it. Also only the first manager actually has the initial global configuration. However, assuming someone uses the embedded runtime, I don't think it's much of a problem, as they will generally have at least one manager running (and they are polluting the runtime with their own modules/config anyway).